### DISCUSSION TITLE
API Key Management Method

### DISCUSSION THREAD

**Discussion by wangfeiing** [emqx/emqx#11319] [Q&A] [answered]
When asking about the HTTP APIs used by different products and services using EMQX, it is necessary to allocate multiple HTTP API keys. If the HTTP API key is passed through EMQX_ API_ KEY__ BOOTSTRAP_ The FILE configuration file cannot be hot updated and requires a restart of the core node to take effect. If an HTTP API key is added through the control panel, the HTTP API keys for each cluster will be different.

How are the API keys managed for different products and services in different clusters?

**Comment by id** [comment]
@wangfeiing you could use https://www.emqx.io/docs/en/v5.1/admin/api-docs.html#tag/API-Keys/paths/~1api_key/post to create API keys.

**Reply by wangfeiing** [reply]
This means that a service needs to connect to different EMQX clusters and requires multiple API Secrets, right?

**Reply by id** [reply, marked as answer]
If a service needs to connect to different EMQX clusters, it will require different API secrets, yes.
